compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2547,9 +2553,12 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                 // Turn trait object paths into `TyKind::TraitObject` instead.
                 match path.res {
                     Res::Def(DefKind::Trait | DefKind::TraitAlias, _) => {
+                        let trait_ref = hir::TraitRef::new(path, hir_id, |span, message| {
+                            self.sess.diagnostic().delay_span_bug(span, message)
```

Comment:
Right now this is unreachable since we just verified that the path resolved to a trait or an alias, but as you said, better safe then sorry :) Might catch a regression in the future.